app/static/src/app/components/WodinPlot.vue

```diff
@@ -97,21 +109,33 @@ export default defineComponent({
 
         let resizeObserver: null | ResizeObserver = null;
 
-        const drawPlot = () => {
+        const drawPlot = (toggleLogScale = false) => {
             if (props.redrawWatches.length) {
                 baseData.value = props.plotData(startTime, props.endTime, nPoints);
 
                 if (hasPlotData.value) {
                     const el = plot.value as unknown;
-                    const layout = {
+                    const layout: Partial<Layout> = {
                         margin,
                         yaxis: {
                             type: yAxisType.value
                         },
                         xaxis: { title: "Time" }
                     };
 
-                    newPlot(el as HTMLElement, baseData.value, layout, config);
+                    const configCopy = { ...config } as Partial<Config>;
+
+                    if (lockYAxis.value && !toggleLogScale) {
```

Comment:
So this means it always does autorange if we're updating because of logScale value change?

Reply:
Yes so the log scale completely changes the axis, if you lock the axis as before there are very weird things y axis values, so instead we don't lock y axis when they toggle between the log and normal scale